README.md

```diff
@@ -1,133 +1,616 @@
+
+Verify the client is able to compile by running the following:
+
+```shell
+cd build/smithyprojections/quickstart-gradle/source/typescript-client-codegen
+# Yarn is used in this example, but equivalent commands using other package managers can be used, e.g. npm and pnpm
+yarn
+yarn build
+```
+
+> Note that running the NPM scripts to verify the generated TypeScript client is NOT part of the code generation process, and needs to be explicitly executed after the client is generated.
 
-6. Run `gradle build` (alternatively, you can use a
-   [Gradle wrapper](https://docs.gradle.org/current/userguide/gradle_wrapper.html)).
+For another example of a Gradle project using Smithy Typescript, the `smithy-typescript-codegen-test` package can be referenced as it builds different TypeScript artifacts through projections.
 
-7. The generated client can be found in `build/smithyprojections/foo-client/source/typescript-codegen`.
+See [the Smithy documentation](https://smithy.io/2.0/guides/building-models/gradle-plugin.html) for more information on build Smithy projects with Gradle.
 
-See [the Smithy documentation](https://smithy.io/2.0/guides/building-models/gradle-plugin.html)
-for more information on build Smithy projects with Gradle.
+> Note: the Smithy Gradle Plugin is under heavy development and is subject to breaking changes.
 
 ## TypeScript code generation
 
-TODO
+By default, the Smithy TypeScript code generators provide the code generation framework to generate TypeScript artifacts (e.g. types, interfaces, implementations) of specified Smithy models. However there are implementations for code generation and TypeScript that either need to be implemented or consumed from third-party packages:
+
+- Protocols: Protocols define how operation shapes (for clients and servers, these are usually inputs and outputs) are serialized and deserialized on the wire. This behavior can be defined in Smithy through [protocol traits](https://smithy.io/2.0/spec/protocol-traits.html) with corresponding implementations of [the `ProtocolGenerator` interface](smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/ProtocolGenerator.java). For example, [AWS SDK for JavaScript v3](https://github.com/aws/aws-sdk-js-v3), a customer of Smithy TypeScript, implements the [AWS protocols](https://smithy.io/2.0/aws/protocols/index.html).
+- Publishing: There is no idiomatic utility to publish generated artifacts since package distribution can vary depending on different technical requirements. For example, [AWS SDK for JavaScript v3](https://github.com/aws/aws-sdk-js-v3), a customer of Smithy TypeScript, has custom tooling to manage versioning, change logs, and publishing in a monorepo.
```

Comment:
Are there any tools we could point customers to to make this easier? Im worried this will be seen as a major adoption barrier if we dont provide some way forward.

Reply:
Pointed to the "Publishing a Client SDK package" which points to some options and how a general publishing workflow could look like.